### DISCUSSION TITLE
RG required in BAM in latest branch release_1.0.2?

### DISCUSSION THREAD

**Discussion by wshands** [GreenleafLab/ArchR#1394] [Questions / Documentation]
We are seeing failures like this
```
2022-04-14 23:15:26 : (BAM_data : 1 of 1) Tabix Bam To Temporary File, 10.091 mins elapsed.

************************************************************
2022-04-14 23:16:02 : ERROR Found in .bamToTmp for (BAM_data : 1 of 1) 
LogFile = ArchRLogs/ArchR-createArrows-15edf0b5b-Date-2022-04-14_Time-23-05-21.log

<simpleError in eval(quote(list(...)), env): object 'RG' not found>

************************************************************
```
I believe when using a bam file with no Read Group in the BAM header as input to CreateArrowFiles in branch release_1.0.2.

Maybe in commit fe4810162e293d98101c277f4cb5ff434280829f the assumption that a RG is in the header was introduced [here](https://github.com/GreenleafLab/ArchR/blob/c61dfca4d346515ce6a40656f8fa1b01cbd73a52/R/CreateArrow.R#L1644) (I'm not an experienced R programmer so I may be way off) 

At least we don't see this problem installing with an earlier commit, i.e.
`    devtools::install_github("GreenleafLab/ArchR@63e7883586a013b9a9c264ddb19dea18d9287087", ref="release_1.0.2", repos = BiocManager::repositories())
`
Would you let us know if this is correct? Thanks!

**Comment by rcorces** [comment]
BAM file input requires specification of where the cellular barcode is stored. This is done using the `bcTag` parameter. Whatever is stored in that BAM field is what gets stored as `RG`. `RG` is not a required field / there is no assumption that RG is in the header. The RG header is created from `bcTag`
https://github.com/GreenleafLab/ArchR/blob/c61dfca4d346515ce6a40656f8fa1b01cbd73a52/R/CreateArrow.R#L1585
